### PR TITLE
added interview read endpoints and matching algo along with tests

### PR DESCRIPTION
**Pull request by elimelt**
Implemented stable matching and added some endpoints, as well as a pretty useful authenticated HTTP client in the test frontend, and a navbar.

### CODE REVIEW

**Comment by elimelt** [comment]
Also feel free to ignore frontend code in review @ShawnCollinge @kipiiler

File: server/interview/algorithm.py
Comment (kipiiler) [inline server/interview/algorithm.py:1]: I think we should define an interface for matching algorithm and maybe implement a strategy pattern if we can create fancier algorithm in the future.

**Review by kipiiler — APPROVED** [review]
otherwise lgtm!

**Comment by kipiiler** [comment]
LGTM

**Pull request merged**